packages/apollo/src/commands/client/push.ts

```diff
@@ -1,70 +1,195 @@
+                  return [
+                    operationHash(
+                      defaultOperationRegistrySignature(document, operationName)
+                    ),
+                    {
+                      operationName,
+                      document,
+                      file: line
+                        ? `${relativePath}:${line}`
+                        : relativePath || ""
+                    }
+                  ];
+                })
+              );
 
-          await project.engine.registerOperations(variables);
+              const { name, referenceID, version } = config.client!;
+              if (!name) {
+                throw new Error("Client name is required to push");
+              }
 
-          // store data for logging
-          ctx.operations = operationManifest;
-          ctx.serviceName = variables.id;
-          ctx.clientIdentity = variables.clientIdentity;
-        }
-      }
-    ]);
+              const variables = {
+                clientIdentity: {
+                  name: name,
+                  identifier: referenceID || name,
+                  version
+                },
+                id: config.name,
+                operations: operationManifest,
+                manifestVersion: 2
+              };
+              const { operations: _op, ...restVariables } = variables;
+              this.debug("Variables sent to Engine:");
+              this.debug(restVariables);
+              this.debug("Operations sent to Engine:");
+              this.debug(operationManifest);
+
+              const {
+                invalidOperations,
+                newOperations,
+                registrationSuccess
+              } = await project.engine.registerOperations(variables);
 
-    this.log(
-      `Successfully pushed ${operations.length} operations from the ${
-        clientIdentity.name
-      } client to the ${serviceName} service in Engine`
-    );
+              if (!registrationSuccess) {
+                if (invalidOperations) {
+                  invalidOperations.forEach(operation => {
+                    const { operationName, file } = signatureToOperation[
+                      operation.signature
+                    ];
+                    result += `\n${chalk.red(
+                      "FAIL"
+                    )}\t${operationName} ${chalk.blue(file)}`;
+                    operation.errors &&
+                      operation.errors.forEach(
+                        ({ message }) => (result += `\n\t${message}`)
+                      );
+                  });
+                  task.title = `Failed to register operations, due to ${
+                    invalidOperations.length
+                  } invalid operations`;
+                  throw Error(invalidOperationsErrorMessage);
+                }
+              } else {
+                if (newOperations && newOperations.length) {
+                  task.title = `Added ${
```

Comment:
Small nit: `Added 6 to Apollo Platform` reads a bit awkward. I'll offer some suggestions, but please feel free to disregard / use your discretion.

`Successfully {registered|pushed|added} 6 operations (optional: to the operation {registry|safelist})`

Reply:
I like it!